semtypes/src/main/java/io/ballerina/types/definition/ListDefinition.java

```diff
@@ -55,37 +64,30 @@ public SemType getSemType(Env env) {
         }
     }
 
-    // Overload define method for commonly used default parameter values
-
-    /***
-     * Define a tuple type without a rest type.
-     */
-    public ComplexSemType define(Env env, List<SemType> initial) {
-        return define(env, initial, initial.size(), PredefinedType.NEVER);
+    public static SemType tuple(Env env, SemType... members) {
+        ListDefinition def = new ListDefinition();
+        return def.define(env, List.of(members), members.length);
     }
 
-    /***
-     * Define a fixed length array type.
-     */
-    public ComplexSemType define(Env env, List<SemType> initial, int size) {
-        return define(env, initial, size, PredefinedType.NEVER);
+    // Overload define method for commonly used default parameter values
+
+    public SemType define(Env env, List<SemType> initial, int size) {
+        return define(env, initial, size, NEVER, CELL_MUT_LIMITED);
     }
 
-    /***
-     * define an array type.
-     */
-    public ComplexSemType define(Env env, SemType rest) {
-        return define(env, new ArrayList<>(), 0, rest);
+    public SemType define(Env env, List<SemType> initial, int fixedLength, SemType rest) {
+        return define(env, initial, fixedLength, rest, CELL_MUT_LIMITED);
     }
 
-    /***
-     * Define a tuple type with a rest type.
-     */
-    public ComplexSemType define(Env env, List<SemType> initial, SemType rest) {
-        return define(env, initial, initial.size(), rest);
+    public SemType define(Env env, List<SemType> initial, int fixedLength, SemType rest,
+                          CellAtomicType.CellMutability mut) {
+        List<CellSemType> initialCells = initial.stream().map(t -> cellContaining(env, t, mut))
+                .toList();
```

Comment:
I think we can write this inline without exceeding 120.

Reply:
Fixed